proc/eval.go

```diff
@@ -177,33 +178,104 @@ func (scope *EvalScope) evalTypeCast(node *ast.CallExpr) (*Variable, error) {
 			return nil, err
 		}
 		typ = &dwarf.PtrType{dwarf.CommonType{int64(scope.Thread.dbp.arch.PtrSize()), exprToString(fnnode)}, ptyp}
+		styp = typ
 	} else {
-		typ, err = scope.findType(exprToString(fnnode))
+		styp, err = scope.findType(exprToString(fnnode))
 		if err != nil {
 			return nil, err
 		}
+		typ = resolveTypedef(styp)
 	}
 
-	// only supports cast of integer constants into pointers
-	ptyp, isptrtyp := typ.(*dwarf.PtrType)
-	if !isptrtyp {
-		return nil, fmt.Errorf("can not convert \"%s\" to %s", exprToString(node.Args[0]), typ.String())
+	converr := fmt.Errorf("can not convert \"%s\" to %s", exprToString(node.Args[0]), typ.String())
+
+	fixargv := func(val constant.Value) *Variable {
```

Comment:
Unless I'm missing `stype` or `typ` getting mutated somewhere else, why not just inline those 2 assignments here and use `argv.Value = val` below instead of calling this function?